Tiled gallery: Add image filters

sirreal opened the pull request:
**Wait until Jetpack 7.1 is near before merging**

Implements #28932

Add a set of image filters to tiled gallery based on #28932.

## To do
- [x] Implement better vertical toolbar like in #28932 
- [x] Restructure to fix the known issue with applying the filter to the selected icon.
  Disabled the filter on selected image
- [x] Finish toolbar styling: https://github.com/Automattic/wp-calypso/pull/30911#issuecomment-466004499
- [x] Sync icons: https://github.com/Automattic/wp-calypso/issues/28932#issuecomment-464798243

## Testing instructions

* Open the block and change the filters.
* Try using the keyboard for the filters.
* Are the filter icons good?
* View the filter on the frontend (requires a WordPress.com patch or Jetpack site testing — gutenpack-jn)

## Screens

### Toolbar

![screen shot 2019-02-22 at 13 10 54](https://user-images.githubusercontent.com/841763/53241875-53805800-36a3-11e9-90a7-47e0dadb9ac8.png)

### Selecting an item with a filter
![filter-workaround](https://user-images.githubusercontent.com/841763/53241982-b07c0e00-36a3-11e9-8f9f-a3299ee7fdb1.gif)


### Original
![orig](https://user-images.githubusercontent.com/841763/52958806-224b1380-3395-11e9-8e1a-91927c9c344f.png)

### Gingham
![gingham](https://user-images.githubusercontent.com/841763/52958800-224b1380-3395-11e9-8b15-45bf70ad0bc9.png)

### Clarendon
![clarendon](https://user-images.githubusercontent.com/841763/52958802-224b1380-3395-11e9-879d-b5245b38c04b.png)

### 1977
![1977](https://user-images.githubusercontent.com/841763/52958804-224b1380-3395-11e9-9d7c-90e1e7d6a731.png)

### Black and white
![bw](https://user-images.githubusercontent.com/841763/52958807-22e3aa00-3395-11e9-9236-dfbf192eb9de.png)

### Sepia
![sep](https://user-images.githubusercontent.com/841763/52958809-22e3aa00-3395-11e9-9dc5-d569e9b6aa9f.png)

sirreal commented:
Unfinished, but FYI: @thomasguillot @simison

simison commented on client/gutenberg/extensions/tiled-gallery/filter-toolbar.js:
```diff
@@ -0,0 +1,139 @@
+								<Rect fill="#f00" x="0" y="0" width="24" height="24" />
+							</SVG>
+						}
+						label={ __( 'Change Text Alignment' ) }
```
>Change Text Alignment

:duck:

sirreal replied:
Whoops. This needs to be changed anyways, I doubt it will be a toolbar at all.

sirreal commented:
I reworked this using the same toolbar/menu as https://github.com/Automattic/wp-calypso/pull/30911#issuecomment-465997441

I'll need to wrap up the item styling on both of these.

I've removed the filter on the selected item, which seems like a good solution.

sirreal commented:
- Rebased
- Updated icons: https://github.com/Automattic/wp-calypso/issues/28932#issuecomment-464798243
- Improved toolbar (matches WordAds https://github.com/Automattic/wp-calypso/pull/30952)

This should be ready!

sirreal commented:
I've labeled this "DO NOT MERGE".

We'll plan to land this around the Jetpack 7.1 release.

scruffian commented:
This is so good. Thanks :)

sirreal commented:
Thanks folks! I'll squash this now so it can easily be picked to 7.1, then we can merge around release.

sirreal commented:
Rebased and squashed. @jeherve this can be picked to a `7.1` branch at any time 👍

eliorivero commented:
@thomasguillot why are the icons needed? Instagram doesn't use icons on their filter, neither does VSCO for example. I don't think I've ever seen an editor that uses icons, they all simply use labels or a preview of the filter applied to a thumbnail.
The 1977 icon is particularly troublesome, it looks redundant. Others like Sepia are confusing, why a flower? IMHO they serve no particular purpose or function and could be suppressed. We would avoid loading all those icons in millions of editors around the world. It's also ecologically greener to remove them than to include them.

<img width="441" alt="icon" src="https://user-images.githubusercontent.com/1041600/53745392-b335ea80-3e7d-11e9-9695-4b93a5cc2654.png">

thomasguillot commented:
@eliorivero looking at the icons again can indeed be troublesome/redundant/confusing.

We should probably switch to generic icons (cc @sirreal):

https://material.io/tools/icons/static/icons/outline-filter_1-24px.svg
https://material.io/tools/icons/static/icons/outline-filter_2-24px.svg
https://material.io/tools/icons/static/icons/outline-filter_3-24px.svg
https://material.io/tools/icons/static/icons/outline-filter_4-24px.svg
https://material.io/tools/icons/static/icons/outline-filter_5-24px.svg
https://material.io/tools/icons/static/icons/outline-filter_none-24px.svg

That being said, we need the icons to keep consistency with Gutenberg's design patterns.

![screenshot 2019-03-04 at 16 19 40](https://user-images.githubusercontent.com/177929/53746571-5ea06880-3e99-11e9-9e1f-791c19f47b40.png)

eliorivero commented:
> That being said, we need the icons to keep consistency with Gutenberg's design patterns.

Ah I see, that's understandable. Thanks for replying 👍

sirreal commented:
@thomasguillot Update filter icons. Now:

![screen shot 2019-03-04 at 18 25 11](https://user-images.githubusercontent.com/841763/53750788-fbb7cd00-3eaa-11e9-879a-fe7510dd4a2a.png)

@jeherve We'll need to pick 384ce09c120027265271220a0c271de077f55f45 into the 7.1 branch

sirreal commented:
@jeherve Another change for 7.1 branch: 427800b71165c88bb0737248631c3e5464032f27

Deselect the selected image when a filter is applied.

Via p1HpG7-6lY-p2 #comment-30815

jeherve commented:
Thank you! Done in 556b30359028842b6802326dd64ba05726bbd90a

sirreal commented:
This was reverted, instead it will be landed as part of #31045
